django/db/models/fields/json.py

```diff
@@ -0,0 +1,537 @@
+                lhs, params = self.preprocess_lhs(compiler, connection, lhs_only=True)
+                lhs = 'JSON_TYPE(%s, %%s)' % lhs
+        return lhs, lhs_params
+
+    def process_rhs(self, compiler, connection):
+        if isinstance(self.rhs, KeyTransform):
+            rhs, rhs_params = super(lookups.Exact, self).process_rhs(compiler, connection)
+            if connection.vendor == 'oracle':
+                rhs_params *= 2
+            return rhs, rhs_params
+        else:
+            rhs, rhs_params = super().process_rhs(compiler, connection)
+        if connection.vendor == 'oracle':
+            func = []
+            for value in rhs_params:
+                val = json.loads(value)
+                if isinstance(val, (list, dict)):
+                    func.append("JSON_QUERY('{\"val\": %s}', '$.val')" % value)
+                else:
+                    func.append("JSON_VALUE('{\"val\": %s}', '$.val')" % value)
+            rhs = rhs % tuple(func)
+            rhs_params = []
+        elif connection.vendor == 'sqlite':
+            func = ["JSON_EXTRACT(%s, '$')" if value != 'null' else '%s' for value in rhs_params]
+            rhs = rhs % tuple(func)
+        return rhs, rhs_params
+
+    def as_oracle(self, compiler, connection):
+        rhs, rhs_params = super().process_rhs(compiler, connection)
+        if rhs_params == ['null']:
+            lhs, lhs_params = self.process_lhs(compiler, connection)
+            prev_lhs, prev_params, key_transforms = self.preprocess_lhs(compiler, connection)
+            json_path = compile_json_path(key_transforms)
+            sql = "(JSON_EXISTS(%s, '%s') AND %s IS NULL)" % (prev_lhs, json_path, lhs)
+            return sql, []
+        else:
+            return super().as_sql(compiler, connection)
+
+
+@KeyTransform.register_lookup
+class KeyTransformIExact(CaseInsensitiveMixin, KeyTransformTextLookupMixin, lookups.IExact):
+    pass
+
+
+@KeyTransform.register_lookup
+class KeyTransformIContains(CaseInsensitiveMixin, KeyTransformTextLookupMixin, lookups.IContains):
+    pass
+
+
+@KeyTransform.register_lookup
+class KeyTransformContains(KeyTransformTextLookupMixin, lookups.Contains):
+    pass
+
+
+@KeyTransform.register_lookup
+class KeyTransformStartsWith(KeyTransformTextLookupMixin, lookups.StartsWith):
+    pass
+
+
+@KeyTransform.register_lookup
+class KeyTransformIStartsWith(CaseInsensitiveMixin, KeyTransformTextLookupMixin, lookups.IStartsWith):
+    pass
+
+
+@KeyTransform.register_lookup
+class KeyTransformEndsWith(KeyTransformTextLookupMixin, lookups.EndsWith):
+    pass
+
+
+@KeyTransform.register_lookup
+class KeyTransformIEndsWith(CaseInsensitiveMixin, KeyTransformTextLookupMixin, lookups.IEndsWith):
+    pass
+
+
+@KeyTransform.register_lookup
+class KeyTransformRegex(KeyTransformTextLookupMixin, lookups.Regex):
+    pass
+
+
+@KeyTransform.register_lookup
+class KeyTransformIRegex(CaseInsensitiveMixin, KeyTransformTextLookupMixin, lookups.IRegex):
```

Comment:
`KeyTransformIRegex` doesn't require `CaseInsensitiveMixin`, it is already handled in 
https://github.com/django/django/blob/4f61810751751b8c5070ce038ea57e949650e9e3/django/db/backends/mysql/operations.py#L305-L314